### PR TITLE
Bump cppcheck action to run on Ubuntu 20.04

### PR DESCRIPTION
**Pull request by turol**
Let's see what breaks....

### CODE REVIEW

**Comment by turol** [comment]
It blew up on the integer overflow checks.

**Comment by turol** [comment]
That might actually be a real issue

**Comment by turol** [comment]
Runner cppcheck is version 1.90

**Comment by turol** [comment]
The change in heretic makes cppcheck shut up by promoting the operation to unsigned (C integer promotion rules suck). However it does generate slightly different assembly. I have not tested this and I'm not sure how to.

File: src/heretic/p_inter.c [hunk 1]
```diff
@@ -1350,7 +1350,7 @@ void P_DamageMobj
         ang = R_PointToAngle2(inflictor->x, inflictor->y,
                               target->x, target->y);
         //thrust = damage*(FRACUNIT>>3)*100/target->info->mass;
-        thrust = damage * (FRACUNIT >> 3) * 150 / target->info->mass;
+        thrust = damage * (FRACUNIT >> 3) * 150u / target->info->mass;
```
Comment (fabiangreffrath) [inline src/heretic/p_inter.c:1353]: Well, I am sure the overflow might indeed happen in Vanilla Heretic. So, by declaring this as unsigned, you fix the bug, but you also change how Heretic works.

**Comment by turol** [comment]
No-one seems to have ideas. Should we just merge it and see if something breaks?

**Comment by fabiangreffrath** [comment]
> No-one seems to have ideas. Should we just merge it and see if something breaks?

If it produces the same ASM, it should be safe.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**